website/pages/blog/open-source-graphql-security.mdx

```diff
@@ -0,0 +1,242 @@
+    flexible plugin system introduced by The Guild for crafting high-quality, performant GraphQL
+    extensions. GraphQL Armor adheres to the ethos of providing accessible, user-friendly security
+    solutions that can be efficiently integrated into your GraphQL setup, safeguarding it from
+    potential vulnerabilities and threats.
+
+With open-source defensive tools like GraphQL Shield and GraphQL Armor, businesses and developers
+can reinforce the security of their GraphQL APIs, protecting data and operations from unauthorized
+access and potential malicious activities. Navigating through the extensive open-source ecosystem
+and leveraging these security tools not only fortifies your GraphQL APIs but also enriches the
+collective knowledge and defense mechanisms of the community, aligning with the spirit of
+collaborative development and security.
+
+In the upcoming section, we’ll delve into offensive tools and resources for learning to equip you
+with a holistic arsenal for safeguarding your GraphQL APIs.
+
+### Offensive Tools: A Proactive Step towards Fortified Defense
+
+When it comes to ensuring an unbreakable security perimeter, adopting an offensive approach is
+pivotal. By putting our defenses to the test and simulating possible attacks, we preemptively expose
+potential vulnerabilities and weaknesses before they can be exploited maliciously. This strategy is
+basically following the idea: the best defense is a good offense. Let’s uncover two powerful
+open-source tools that allow us to proactively secure our GraphQL APIs by adopting an offensive
+stance.
+
+1.  **[Goctopus](https://escape.tech/blog/introducing-goctopus/)**
+
+    Say hello to Goctopus, our open-source GraphQL endpoint discovery and fingerprinting tool. This
+    tentacled marvel meticulously hunts down GraphQL endpoints, fingerprinting their authentication
+    methods and schema availability, ensuring not a single detail slips through its grasp. Built by
+    the Escape team, Goctopus lends developers and businesses a watchful eye, enabling them to
+    identify and comprehend the specifics of GraphQL endpoints with precision, ensuring that they
+    can reinforce any possible points of vulnerability within their organization before they become
+    a target.
+
+2.  **[Clairvoyance](https://github.com/nikitastupin/clairvoyance)**
+
+    Meet Clairvoyance, a savvy tool designed to peek into GraphQL APIs even when introspection is
+    disabled, which is a common security measure (e.g.,
+    [Apollo Server automatically disables introspection in production environments](https://www.apollographql.com/docs/tutorial/schema/#explore-your-schema)).
+    Clairvoyance subtly and skillfully retrieves schema information from these APIs, providing
+    developers with insights into the API structure and potential vulnerability points. This becomes
+    particularly vital when dealing with APIs that have sought to shield their schema details as a
+    security measure.
```

Comment:
@EmrysMyrddin I know we had this discussion but should we do the same in Yoga? (disabled by default)

Reply:
If I can give my pov here, I think having a setting and mentioning it in the docs is primordial, but disabling it isn't. Security by obscurity is never an optimal solution, and brute-forcing routes is doable even with a closed introspection & disabled suggestions :)